Realm/RLMResults.mm

```
@@ -328,6 +336,7 @@ - (RLMResults *)objectsWithPredicate:(NSPredicate *)predicate {
         if (_results.get_mode() == Results::Mode::Empty) {
             return self;
         }
+        // FIXME: primitive array queries
```

Should we throw for now rather than silently giving incorrect results?